apps/internal/oauth/resolvers.go

```diff
@@ -123,13 +123,15 @@ func (m *authorityEndpoint) addCachedEndpoints(authorityInfo authority.Info, use
 }
 
 func (m *authorityEndpoint) openIDConfigurationEndpoint(ctx context.Context, authorityInfo authority.Info, userPrincipalName string) (string, error) {
-	if authorityInfo.ValidateAuthority && !authority.TrustedHost(authorityInfo.Host) {
+	if authorityInfo.Tenant != "adfs" && authorityInfo.ValidateAuthority && !authority.TrustedHost(authorityInfo.Host) {
 		resp, err := m.rest.Authority().AADInstanceDiscovery(ctx, authorityInfo)
 		if err != nil {
 			return "", err
 		}
 
 		return resp.TenantDiscoveryEndpoint, nil
+	} else if authorityInfo.Tenant == "adfs" {
```

Comment:
If you make this this first condition in the `if` block you can remove the `!=` check that's currently on line 126.